backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -154,12 +151,11 @@ private async void OnEmergencyButtonPressedForRobot(object? sender, EmergencyBut
                 return;
             }
 
-            await MissionSchedulingService.FreezeMissionRunQueueForRobot(e.RobotId);
+            try { await MissionSchedulingService.FreezeMissionRunQueueForRobot(e.RobotId); }
+            catch (RobotNotFoundException) { return; }
```

Comment:
I am confused, we read the robot from the database and it wasn't null, why do we need this catch?

Reply:
With the current db inconsistencies it might be possible for the robot to be deleted during the execution of this event handler, so there's no harm in having this try/catch in my opinion.